perception/image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp

```diff
@@ -116,16 +131,21 @@ void SegmentPointCloudFusionNode::fuseOnSingleImage(
       static_cast<uint16_t>(normalized_projected_point.y()),
       static_cast<uint16_t>(normalized_projected_point.x()));
     if (static_cast<size_t>(semantic_id) >= filter_semantic_label_target_.size()) {
-      output_cloud.push_back(pcl::PointXYZ(*iter_orig_x, *iter_orig_y, *iter_orig_z));
+      copyPointCloud(
+        input_pointcloud_msg, point_step, global_offset, output_cloud, output_pointcloud_size);
       continue;
     }
     if (!filter_semantic_label_target_.at(semantic_id)) {
-      output_cloud.push_back(pcl::PointXYZ(*iter_orig_x, *iter_orig_y, *iter_orig_z));
+      copyPointCloud(
+        input_pointcloud_msg, point_step, global_offset, output_cloud, output_pointcloud_size);
     }
   }
 
-  pcl::toROSMsg(output_cloud, output_pointcloud_msg);
-  output_pointcloud_msg.header = input_pointcloud_msg.header;
+  output_cloud.data.resize(output_pointcloud_size);
+  output_cloud.row_step = output_pointcloud_size / output_cloud.height;
+  output_cloud.is_bigendian = input_pointcloud_msg.is_bigendian;
+  output_cloud.is_dense = input_pointcloud_msg.is_dense;
```

Comment:
Any specific reason why these two lines are declared nearby the following (previous) lines?
```cpp
  output_cloud.fields = input_pointcloud_msg.fields;
  output_cloud.header = input_pointcloud_msg.header;
  output_cloud.height = input_pointcloud_msg.height;
  output_cloud.point_step = input_pointcloud_msg.point_step;
```

Reply:
@kminoda No, so I changed as your suggestion. https://github.com/autowarefoundation/autoware.universe/pull/7224/commits/61a137dbe6417424cabaa5a5a986227afa9dd348